packages/calcite-components/src/components/inline-editable/inline-editable.tsx

```diff
@@ -305,28 +305,29 @@ export class InlineEditable
   //
   //--------------------------------------------------------------------------
 
-  mutationObserverCallback(): void {
-    this.updateSlottedInput();
-    this.scale = this.scale || this.inputElement?.scale;
-  }
-
-  onLabelClick(): void {
-    this.setFocus();
-  }
-
-  updateSlottedInput(): void {
-    const inputElement: HTMLCalciteInputElement = getSlotted(this.el, {
-      matches: "calcite-input",
-    });
+  handleDefaultSlotChange = (event: Event): void => {
+    const inputElement = slotChangeGetAssignedElements(event).filter(
+      (el): el is HTMLCalciteInputElement => el.matches("calcite-input"),
+    )[0];
 
     this.inputElement = inputElement;
 
     if (!inputElement) {
       return;
     }
 
-    this.inputElement.disabled = this.disabled;
-    this.inputElement.label = this.inputElement.label || getLabelText(this);
+    inputElement.disabled = this.disabled;
+    inputElement.label = inputElement.label || getLabelText(this);
+  };
+
+  mutationObserverCallback(): void {
+    // todo: kind of odd to be setting scale based on a slotted element scale.
+    // would require a scale event to properly handle this which would be better than a mutation observer
+    this.scale = this.scale || this.inputElement?.scale;
```

Comment:
Its kind of odd to be setting scale based on a slotted element's scale.

To handle this better, an internal event for scale change would be needed.

IMO we should just remove this. Thoughts?

Reply:
The problem with this is that if inputElements scale changes after being slotted, this wouldn't get picked up.
If inputElement is replaced after initialization then the scale is already set at that point too.

Reply:
Agreed. Can you create an issue for this? 

@geospatialem @brittneytewks We can plan for removal in a breaking change.

Reply:
https://github.com/Esri/calcite-design-system/issues/10466